### PR TITLE
Grouping examples into categories

### PR DESCRIPTION
**Pull request by Taha-Hassan-Git**
This PR adds collapsible groups to the examples app.

it's not finished, but I'd like a review before continuing as I've made a few decisions I'd like feedback on. I'd like to make a separate issue for abstracting the accordion component, as I wasn't sure how I would do it and I thought it would be best to prioritise the functionality first. Especially considering there are more pressing issues to be getting on with.

### Change Type

- [ ] `patch` — Bug fix
- [ ] `minor` — New feature
- [ ] `major` — Breaking change
- [ ] `dependencies` — Changes to package dependencies[^1]
- [x] `documentation` — Changes to the documentation only[^2]
- [ ] `tests` — Changes to any test code only[^2]
- [ ] `internal` — Any other changes that don't affect the published package[^2]
- [ ] I don't know

[^1]: publishes a `patch` release, for devDependencies use `internal`
[^2]: will not publish a new version

### Release Notes

- Add collapsible categories to the examples app

### CODE REVIEW

File: apps/examples/src/ExamplePage.tsx
Comment (mimecuvalo) [inline apps/examples/src/ExamplePage.tsx:89]: feels like this isn't the best place to put this - but i'm not familiar yet with a better place in the codebase to put this? i guess it's fine for now ¯\_(ツ)_/¯

Comment (mimecuvalo) [inline apps/examples/src/ExamplePage.tsx:89]: feels like this isn't the best place to put this - but i'm not familiar yet with a better place in the codebase to put this? i guess it's fine for now ¯\\_(ツ)_/¯

File: apps/examples/src/ExamplePage.tsx
Comment (mimecuvalo) [inline apps/examples/src/ExamplePage.tsx:69]: nit: i know this was like this before but it'd be better to avoid a short name like `e` because it can better to avoid shortnames to improve readability. so maybe something like `sidebarExample`? (open to other ideas)

File: apps/examples/src/ExamplePage.tsx
Comment (mimecuvalo) [inline apps/examples/src/ExamplePage.tsx:57]: nit: (similar to below), it'd be better to have `category` vs `c` - easier to read the code and also easier to grep through when looking for where the variable is used

File: apps/examples/src/HomePage.tsx
Comment (mimecuvalo) [inline apps/examples/src/HomePage.tsx:1]: i think we should just kill this page but that's neither here nor there :)

File: apps/examples/src/HomePage.tsx
Comment (mimecuvalo) [inline apps/examples/src/HomePage.tsx:18]: this logic doesn't apply anymore?

File: apps/examples/src/components/SpanLink.tsx
Comment (mimecuvalo) [inline apps/examples/src/components/SpanLink.tsx:9]: may i suggest actually just calling this Link?
in the future it might not be a `span` (but, let's say a div) and we wouldn't want to rename this again :)

File: apps/examples/src/examples.tsx
Comment (mimecuvalo) [inline apps/examples/src/examples.tsx:10]: i don't think null should be a category option? is there a good reason for that?

File: apps/examples/src/examples.tsx
Comment (mimecuvalo) [inline apps/examples/src/examples.tsx:16]: so, for this logic below, what would be better is to consolidate it (make it so when adding a category/removing a category, you don't have to think of another place where you have to add a codeblock/maintain extra code to make something work)

we should make it more like:
```
enum CATEGORIES = {
  BASIC = 'basic',
  EDITOR = 'editor'
  etc...
}

const CATEGORY_TO_TITLE = {
  BASIC = 'Getting Started',
  EDITOR = 'Editor API'
  etc...
}

export const examples = CATEGORIES.map(category => {
  Object.values(import.meta.glob('./examples/*/README.md', { eager: true })) as Example[]
)
	.filter((e) => e.category === category)
  ...

  return { id: CATEGORY_TO_TITLE[category], values: ...}
})
```
(also, above in the type, you can do something like: `category: CATEGORIES` instead of listing it out again)

File: apps/examples/src/examples.tsx
Comment (mimecuvalo) [inline apps/examples/src/examples.tsx:100]: this is touched upon above but `array` is too specific on what basicExamples is. it might be that basicExamples becomes an object in the future. a better wording here would be `values` because it might change.

File: apps/examples/src/styles.css
Comment (mimecuvalo) [inline apps/examples/src/styles.css:89]: nit: (does prettier take care of this?) should be a space before `{` (and below)

File: apps/examples/src/styles.css
Comment (mimecuvalo) [inline apps/examples/src/styles.css:104]: you don't need to do the browser prefixes anymore
(i'm assuming that we're using postcss or something as the processor here, i could be wrong) - but you just need to do `transition`

File: apps/examples/src/styles.css
Comment (mimecuvalo) [inline apps/examples/src/styles.css:129]: would love to see these colors in a consolidated variable somewhere but i won't press you on this if it's not really strictly done in this app

File: apps/examples/src/examples/api/README.md
Comment (mimecuvalo) [inline apps/examples/src/examples/api/README.md:5]: did i miss where these `order`s are being used? maybe you pulled it out later?

in any case, it's a moot point: we shouldn't have `order` - if you pull an example or add a new one, you would have to change all the numbers in several files - no fun :P

we can talk about different approaches on this but, either we have maybe a `priority` option where some things bubble up higher or lower in the list, or, i guess, alphabetize.

**Review by mimecuvalo — COMMENTED** [review]
great start @Taha-Hassan-Git ! i think the accordion feels great :)

couple comments — lemme know if you have any questions!

File: apps/examples/src/HomePage.tsx
Comment (Taha-Hassan-Git) [reply apps/examples/src/HomePage.tsx:18]: There are one or two hidden examples, not sure why they've been hidden though.

File: apps/examples/src/styles.css
Comment (Taha-Hassan-Git) [reply apps/examples/src/styles.css:89]: Weirdly, it doesn't. Might have to ask someone about this.

File: apps/examples/src/examples/api/README.md
Comment (Taha-Hassan-Git) [reply apps/examples/src/examples/api/README.md:5]: I forgot to sort them! I was going to order them all according to their level of complexity, but you raise a good point about maintainability. I think I'll go for priority, a number between 1 and 3 that communicates how complex or difficult the example is.

File: apps/examples/src/HomePage.tsx
Comment (mimecuvalo) [reply apps/examples/src/HomePage.tsx:18]: Oh, ok - then we should keep this logic around?

File: .github/ISSUE_TEMPLATE/request_example.yml
Comment (mimecuvalo) [inline .github/ISSUE_TEMPLATE/request_example.yml:1]: i think this might have leaked out from your other PR ( https://github.com/tldraw/tldraw/pull/2597 ) :) let's remove

File: apps/examples/vite.config.ts
Comment (mimecuvalo) [inline apps/examples/vite.config.ts:87]: minor: we should get rid of `priority !== null`
and prbly same goes for category check below too

File: apps/examples/src/styles.css
Comment (mimecuvalo) [inline apps/examples/src/styles.css:4]: niiiice

File: apps/examples/src/examples/canvas-events/CanvasEventsExample.tsx
Comment (mimecuvalo) [inline apps/examples/src/examples/canvas-events/CanvasEventsExample.tsx:13]: this is a great start! already better than what we had - let's split this change into a separate PR - i think it still needs a little more work and I don't wanna hold up this larger change :)

**Review by mimecuvalo — APPROVED** [review]
pre-approving modulo the latest comments before you land please!

File: apps/examples/src/examples/canvas-events/CanvasEventsExample.tsx
Comment (Taha-Hassan-Git) [reply apps/examples/src/examples/canvas-events/CanvasEventsExample.tsx:13]: This leaked as well!

**Pull request merged**
